documentation/design-docs/eventcounters.md

```diff
@@ -0,0 +1,306 @@
+
+And that's it! For `PollingCounter` (or `IncrementingPollingCounter`), there is nothing else that needs to be done since they poll the values themselves at the interval requested by the consumer.
+
+Here is another example of runtime counter implemented using `IncrementingPollingCounter`.
+
+```cs
+IncrementingPollingCounter monitorContentionCounter = new IncrementingPollingCounter(
+    "monitor-lock-contention-count",
+    this,
+    () => Monitor.LockContentionCount
+)
+{
+    DisplayName = "Monitor Lock Contention Count",
+    DisplayRateTimeScale = new TimeSpan(0, 0, 1)
+};
+```
+
+This counter uses the [Monitor.LockContentionCount](https://docs.microsoft.com/en-us/dotnet/api/system.threading.monitor.lockcontentioncount?view=netcore-3.0) API to report the increment of the total lock contention count. The `DisplayRateTimeScale` property is an optional `TimeSpan` which can be set to provide a hint of what time interval this counter is best displayed at. For example, the lock contention count is best displayed as *count per second*, so its `DisplayRateTimeScale` is set to 1 second. This can be adjusted for different types of rate counters.
+
+There are more runtime counter implementation to use as a reference in the [CoreCLR](https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/src/System/Diagnostics/Eventing/RuntimeEventSource.cs) repo.
+
+Here is a sample code for a full `EventSource` implementation that uses `EventCounter`:
+```cs
+// Give your event sources a descriptive name using the EventSourceAttribute, otherwise the name of the class is used. 
+[EventSource(Name = "Samples-EventCounterDemos-Minimal")]
+public sealed class MinimalEventCounterSource : EventSource
+{
+    // define the singleton instance of the event source
+    public static MinimalEventCounterSource Log = new MinimalEventCounterSource();
+    private EventCounter requestTimeCounter;
+
+    private MinimalEventCounterSource() : base(EventSourceSettings.EtwSelfDescribingEventFormat) 
+    {
+        this.requestTimeCounter = new EventCounter("request-time", this)
+        {
+            DisplayName = "Request Processing Time",
+            DisplayUnit = "MSec"
+        }
+    }
+
+    /// <summary>
+    /// Call this method to indicate that a request for a URL was made which took a particular amount of time
+    public void Request(string url, float elapsedMSec)
+    {
+        this.requestTimeCounter.WriteMetric(elapsedMSec);
+    }
+}
+```
+
+## Concurrency 
+It is important to note that the EventCounters API does not guarantee any sort of thread safety. It is the author's responsibility to guarantee the thread-safety of the data being passed to the counter APIs.
```

Comment:
This statement feels confusing. The WriteMetric and Increment APIs are thread-safe, and the callbacks are already synchronized (ie EventCounters API _does_ guarantee thread safety for the APIs that would likely be used in multi-threaded code). This section is referring to the function Request() which is code the user hypothetically authored. I'd like to think it can go without saying that if the user wants to write their own functions that will called from multiple threads concurrently then they are responsible for ensuring the thread safety of the code they write.